src/voice-broadcast/audio/VoiceBroadcastRecorder.ts

```diff
@@ -0,0 +1,140 @@
+    private pagesFromRecorderCount = 0;
+
+    public constructor(
+        private voiceRecording: VoiceRecording,
+        public readonly targetChunkLength: number,
+    ) {
+        this.voiceRecording.onDataAvailable = this.onDataAvailable;
+    }
+
+    public async start(): Promise<void> {
+        return this.voiceRecording.start();
+    }
+
+    /**
+     * Stops the recording and returns the remaining chunk (if any).
+     */
+    public async stop(): Promise<ChunkRecordedPayload> {
+        await this.voiceRecording.stop();
+        return this.extractChunk();
+    }
+
+    public on(event: string | symbol, listener: (...args: any[]) => void): this {
+        this.voiceRecording.on(event, listener);
+        return this;
+    }
+
+    public off(event: string | symbol, listener: (...args: any[]) => void): this {
+        this.voiceRecording.off(event, listener);
+        return this;
+    }
+
+    public get contentType(): string {
+        return this.voiceRecording.contentType;
+    }
+
+    private onDataAvailable = (data: ArrayBuffer) => {
+        const dataArray = new Uint8Array(data);
+        this.pagesFromRecorderCount++;
+
+        if (this.pagesFromRecorderCount <= 2) {
+            // first two pages contain the headers
+            this.headers = concat(this.headers, dataArray);
+            return;
+        }
+
+        this.handleData(dataArray);
+    };
+
+    private handleData(data: Uint8Array) {
+        this.chunkBuffer = concat(this.chunkBuffer, data);
+        this.emitChunkIfTargetLengthReached();
+    }
+
+    private emitChunkIfTargetLengthReached() {
+        if (this.chunkLength >= this.targetChunkLength) {
+            this.emitAndResetChunk();
+        }
+    }
+
+    /**
+     * Extracts the current chunk and resets the buffer.
+     */
+    private extractChunk(): ChunkRecordedPayload {
+        if (this.chunkBuffer.length === 0) {
+            return null;
+        }
+
+        const currentRecorderTime = this.voiceRecording.recorderSeconds;
+        const payload: ChunkRecordedPayload = {
+            buffer: concat(this.headers, this.chunkBuffer),
+            length: this.chunkLength,
+        };
+        this.chunkBuffer = new Uint8Array(0);
+        this.previousChunkEndTimePosition = currentRecorderTime;
+        return payload;
+    }
+
+    private emitAndResetChunk() {
+        if (this.chunkBuffer.length === 0) {
+            return;
+        }
+
+        this.voiceRecording.emit(
+            VoiceBroadcastRecorderEvent.ChunkRecorded,
+            this.extractChunk(),
+        );
+    }
+
+    private get chunkLength() {
+        return this.voiceRecording.recorderSeconds - this.previousChunkEndTimePosition;
+    }
+
+    public destroy(): void {
+        this.voiceRecording.destroy();
+    }
+}
+
+export const createVoiceBroadcastRecorder = () => {
+    const targetChunkLength = SdkConfig.get("voice_broadcast")?.chunk_length || DEFAULTS.voice_broadcast.chunk_length;
```

Comment:
```suggestion
    const targetChunkLength = SdkConfig.get("voice_broadcast")?.chunk_length || DEFAULTS.voice_broadcast?.chunk_length || (60 * 1000);
```

because `voice_broadcast` is purely optional

Reply:
I don't get the point of `DEFAULTS` then, if we add another defaults everywhere.
It would be possible to export the default value from `DEFAULTS` as well, but does it make sense?
Tests should currently check, that it uses the `DEFAULTS` value.

Reply:
yea, after I made this comment I reviewed something else which was relying on `DEFAULTS` being populated - I'd say that's safe enough then.